src/code_path.ml

```diff
@@ -10,9 +10,14 @@ type t = {
   in_expr : bool;
 }
 
+let remove_all_extensions basename =
+  match String.split_on_char ~sep:'.' basename with
+  | [] -> assert false
```

Comment:
Just noting that `Stdlib.Filename.remove_extension` doesn't raise in the case where there is no extension, it just passes the filename through. Not sure if we want to do the same here or not.

Reply:
This won't raise as this is an invariant of `String.split_on_char`, see: https://ocaml.org/manual/5.2/api/String.html#extract.

I'll add a comment here, you're right that it was a bit bold to just dump an `assert false` here!

Reply:
The more you know! Thanks :))